Release notes 2.16.9 (#12587)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## Release 2.16.9

### `prefect deploy` with `-jv/--job-variable` option

In a prior release, `prefect deployment run` added `-jv/--job-variable` for providing job variables. We want to be consistent in our CLI by adding a `prefect deploy` `-jv/--job-variable` option as well! We have deprecated the `-v/--variables` option.

See the following pull request for implementation details:
- https://github.com/PrefectHQ/prefect/pull/12410

### Enhancements
- Remove nested task constraint that prevented tasks called from other tasks — https://github.com/PrefectHQ/prefect/pull/12548
- Stop creating artifacts for unpersisted results - https://github.com/PrefectHQ/prefect/pull/12454
- Allow for deletion of work pool workers via API — https://github.com/PrefectHQ/prefect/pull/12330

> - Raise nicer error on `prefect worker start -t bad-type` - https://github.com/PrefectHQ/prefect/pull/12586

Reviewer: ```suggestion
- Raise nicer error on `prefect worker start -t bad-type` - https://github.com/PrefectHQ/prefect/pull/12586
- Add tooltip to variable names in the ui and increase width given for displaying variable names https://github.com/PrefectHQ/prefect-ui-library/pull/2275
```

### Fixes
- Raise lower bound on `typer` dependency — https://github.com/PrefectHQ/prefect/pull/12512
- Skip flow run cancellation if no associated deployment — https://github.com/PrefectHQ/prefect/pull/12001
- Handle referenced blocks in base templates during `job_variable` validation — https://github.com/PrefectHQ/prefect/pull/12329
- Select correct `AsyncWaiter` for successively awaited flow and task calls — https://github.com/PrefectHQ/prefect/pull/12510
- Handle flow run creation for runner-managed deployments — https://github.com/PrefectHQ/prefect/pull/12319
- Expose `ignore_warnings` in `Flow.deploy` — https://github.com/PrefectHQ/prefect/pull/12569
- Allow `prefect cloud login` re-authentication in non-interactive mode — https://github.com/PrefectHQ/prefect/pull/12575

> - Update ECS provisioner IAM policy to include `ecs:TagResource` permission — https://github.com/PrefectHQ/prefect/pull/12551

Reviewer: ```suggestion
- Update ECS provisioner IAM policy to include `ecs:TagResource` permission — https://github.com/PrefectHQ/prefect/pull/12551
- Correctly populate default parameters in the create flow run form https://github.com/PrefectHQ/prefect-ui-library/pull/2280
```

# Prefect Release Notes

## Release 2.16.9

### `prefect deploy` with `-jv/--job-variable` option

> In a prior release, `prefect deployment run` added `-jv/--job-variable` for providing job variables. We want to be consistent in our CLI by adding a `prefect deploy` `-jv/--job-variable` option as well! We have deprecated the `-v/--variables` option.

Author: ```suggestion
In a prior release, we added a `-jv/--job-variable` for providing job variables when running a deployment via the CLI using `prefect deployment run`. We want to be consistent in our CLI by adding a `-jv/--job-variable` option while creating deployments as well by adding this to our `prefect deploy` command! We have deprecated the `-v/--variables` option.
```

> In a prior release, we added a `-jv/--job-variable` for providing job variables when running a deployment using `prefect deployment run`. We want to be consistent in our CLI by allowing you to use this option while creating deployments during `prefect deploy`! Thus, we have added a `-jv/--job-variable` option to `prefect deploy` to replace the `-v/--variables` option, which we have now deprecated.

Author: @zzstoatzz how does this sound?

Reviewer 2: 💯

> In a prior release, we added a `-jv/--job-variable` for providing job variables when running a deployment using `prefect deployment run`. We want to be consistent in our CLI by allowing you to use this option while creating deployments during `prefect deploy`! Thus, we have added a `-jv/--job-variable` option to `prefect deploy` to replace the `-v/--variables` option, which we have now deprecated.

Author: ```suggestion
In a prior release, we added a `-jv/--job-variable` option for providing job variables when running a deployment using `prefect deployment run`. We want to be consistent in our CLI by allowing you to use this option while creating deployments during `prefect deploy`! Thus, we have added a `-jv/--job-variable` option to `prefect deploy` to replace the `-v/--variables` option, which we have now deprecated.
```

See the following pull request for implementation details:
- https://github.com/PrefectHQ/prefect/pull/12410

### Enhancements
- Remove nested task constraint that prevented tasks called from other tasks — https://github.com/PrefectHQ/prefect/pull/12548
- Stop creating artifacts for unpersisted results - https://github.com/PrefectHQ/prefect/pull/12454
- Allow for deletion of work pool workers via API — https://github.com/PrefectHQ/prefect/pull/12330

> - Raise nicer error on `prefect worker start -t bad-type` - https://github.com/PrefectHQ/prefect/pull/12586

Reviewer 2: ```suggestion
- Raise more informative error on `prefect worker start -t bad-type` - https://github.com/PrefectHQ/prefect/pull/12586
```

- Add tooltip to variable names in the ui and increase width given for displaying variable names https://github.com/PrefectHQ/prefect-ui-library/pull/2275

### Fixes
- Raise lower bound on `typer` dependency — https://github.com/PrefectHQ/prefect/pull/12512
- Skip flow run cancellation if no associated deployment — https://github.com/PrefectHQ/prefect/pull/12001
- Handle referenced blocks in base templates during `job_variable` validation — https://github.com/PrefectHQ/prefect/pull/12329
- Select correct `AsyncWaiter` for successively awaited flow and task calls — https://github.com/PrefectHQ/prefect/pull/12510
- Handle flow run creation for runner-managed deployments — https://github.com/PrefectHQ/prefect/pull/12319
- Expose `ignore_warnings` in `Flow.deploy` — https://github.com/PrefectHQ/prefect/pull/12569
- Allow `prefect cloud login` re-authentication in non-interactive mode — https://github.com/PrefectHQ/prefect/pull/12575
- Update ECS provisioner IAM policy to include `ecs:TagResource` permission — https://github.com/PrefectHQ/prefect/pull/12551

> - Correctly populate default parameters in the create flow run form https://github.com/PrefectHQ/prefect-ui-library/pull/2280

Author: ```suggestion
- Correctly populate custom default parameters in the flow submission form https://github.com/PrefectHQ/prefect-ui-library/pull/2280
```
